### DISCUSSION TITLE
Auth on client-side is working but not on server-side

### DISCUSSION THREAD

**Discussion by aaronyih1** [gladly-team/next-firebase-auth#120] [Q&A] [answered]
Client side auth functions are working as expected, but Auth cookies are not being set. My cookies config is as follows: 
```
cookies: {
      name: 'CanMan', // required
      // Keys are required unless you set `signed` to `false`.
      // The keys cannot be accessible on the client side.
      keys: [
        process.env.COOKIE_SECRET_CURRENT,
        process.env.COOKIE_SECRET_PREVIOUS,
      ],
      httpOnly: true,
      maxAge: 12 * 60 * 60 * 24 * 1000, // twelve days
      overwrite: true,
      path: '/',
      sameSite: 'strict',
      secure: false, // set this to false in local (non-HTTPS) development
      signed: false,
    },
```

Logging the AuthUser in getServerSideProps as shown below returns null:

```
export const getServerSideProps = withAuthUserSSR()(
async ({AuthUser})=>
  {
    console.log(AuthUser)
    return { props: {} }
  }
)
```

```
{
  id: null,
  email: null,
  emailVerified: false,
  getIdToken: [Function],
  clientInitialized: false,
  firebaseUser: null,
  signOut: [Function],
  serialize: [Function: serialize]
}
```

**Comment by kmjennison** [comment]
Have you tried [these troubleshooting steps](https://github.com/gladly-team/next-firebase-auth#server-side-auth-is-not-working-the-user-and-token-are-always-null-when-using-withauthusertokenssr-but-client-side-auth-works)? Can you confirm the auth cookies are set and are sent along with requests?

**Reply by aaronyih1** [reply]
I did, I looked for a CanMan cookie in the application panel of the Chrome Dev tools, but didn't see anything there. It seems like the cookies are not being set at all.

**Reply by kmjennison** [reply]
Do you see a call to your login API endpoint? Does it return a `set-cookie` header?

**Reply by aaronyih1** [reply]
I do. It is getting called. Here's my login endpoint.
```
// ./pages/api/login
import { setAuthCookies } from 'next-firebase-auth'

const handler = async (req, res) => {
  try {
    await setAuthCookies(req, res)
  } catch (e) {
    // return res.status(500).json({ error: e })
  }
  return res.status(200).json({ success: true })
}

export default handler
```

I'm not seeing a `set-cookie` header. All I see is a `Cookie` header that shows all the cookies, but not my CanMan one.

**Reply by aaronyih1** [reply]
Uncommenting the error log gets me this error, which leads me to believe that I'm including the private key improperly. 

> {"error":{"code":"app/invalid-credential","message":"Service account object must contain a string \"private_key\" property."}}

I'm simply including it as described in the Readme. I put it in .env.local with double quotes:
`FIREBASE_PRIVATE_KEY='"-----BEGIN PRIVATE KEY-----[KEY HERE]-----END PRIVATE KEY-----\n"'`

What am I doing wrong??

**Comment by aaronyih1** [comment, marked as answer]
I discovered that I had misnamed my env file `env.local` instead of `.env.local`. This was preventing my secret key from being available to firebase admit init.

**Reply by kmjennison** [reply]
Thanks for following up. Added this line to the README troubleshooting: "In addition, please double-check your server logs for any errors to ensure the Firebase admin app is initializing properly."
